ts/hooks/use_wallet.ts

```diff
@@ -63,45 +64,59 @@ export const useWallet = () => {
         }
     }, [currentProviderState.provider]);
 
-    const connectToWallet = useCallback(async () => {
-        cleanupCurrentProvider();
-        const providerState = providerStateFactory.getInitialProviderState(currentNetworkId);
+    const connectToWallet = useCallback(
+        async (providerOverride?: Providers) => {
+            cleanupCurrentProvider();
+            const providerState: ProviderState =
+                providerOverride === Providers.WalletLink
+                    ? providerStateFactory.getInitialProviderStateFromWalletLink(currentNetworkId)
+                    : providerStateFactory.getInitialProviderState(currentNetworkId);
 
-        const networkId = await providerState.web3Wrapper.getNetworkIdAsync();
-        if (networkId !== currentNetworkId) {
-            dispatcher.updateNetworkId(networkId);
-        }
+            dispatcher.updateProviderState(providerState);
```

Comment:
This little bug was pretty tricky to hunt down :feelsgood: Previously we would set the `providerState` when initializing the redux store and then never update it. That meant that it was either set to use MetaMask or a read-only RPC provider in the case where there was no MM.

When connecting to another wallet like Wallet Link it would successfully initialize the new provider and providerState but never update the value in the store so when you later try sending a transaction it's using the wrong provider.

Reply:
oh gosh